src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -516,14 +523,44 @@ private void CreateRefPath(ODataPath currentPath)
         /// <param name="convertSettings">The settings for the current conversion.</param>
         private void CreateCountPath(ODataPath currentPath, OpenApiConvertSettings convertSettings)
         {
-            if(currentPath == null) throw new ArgumentNullException(nameof(currentPath));
-            if(convertSettings == null) throw new ArgumentNullException(nameof(convertSettings));
-            if(!convertSettings.EnableDollarCountPath) return;
+            Utils.CheckArgumentNull(currentPath, nameof(currentPath));
+            Utils.CheckArgumentNull(convertSettings, nameof(convertSettings));
+
+            if(!convertSettings.EnableDollarCountPath) 
+                return;
+
             var countPath = currentPath.Clone();
             countPath.Push(ODataDollarCountSegment.Instance);
             AppendPath(countPath);
         }
 
+        /// <summary>
+        /// Create path with alternate key
+        /// </summary>
+        /// <param name="currentPath">The current OData path.</param>
+        /// <param name="entityType">The entityType with alternate keys</param>
+        private void CreateAlternateKeyPath(ODataPath currentPath, IEdmEntityType entityType)
+        {
+            Utils.CheckArgumentNull(currentPath, nameof(currentPath));
+            Utils.CheckArgumentNull(entityType, nameof(entityType));
+
+            IEnumerable<IDictionary<string, IEdmProperty>> alternateKeys = _model.GetAlternateKeysAnnotation(entityType);
+            if (alternateKeys.Count() > 0)
+            {
+                foreach (var keyDict in alternateKeys)
```

Comment:
calling .Count() and do foreach will perform double evaluation for the IEnumerable< T >. Maybe consider to call .ToList() or .ToArray() first?